rmgpy/yml.py

```diff
@@ -0,0 +1,168 @@
+import yaml
+import os
+from rmgpy.chemkin import loadChemkinFile
+from rmgpy.species import Species
+from rmgpy.reaction import Reaction
+from rmgpy.thermo.nasa import NASAPolynomial, NASA
+from rmgpy.thermo.wilhoit import *
+from rmgpy.kinetics.arrhenius import *
+from rmgpy.kinetics.falloff import *
+from rmgpy.kinetics.chebyshev import *
+from rmgpy.data.solvation import *
+from rmgpy.util import makeOutputSubdirectory
+
+def convertchemkin2yml(chemkinpath,spcdictpath=None,output="chem.rms"):
+    if spcdictpath:
+        spcs,rxns = loadChemkinFile(chemkinpath,dictionaryPath=spcdictpath)
+    else:
+        spcs,rxns = loadChemkinFile(chemkinpath)
+    writeyml(spcs,rxns,path=output)
+
+def writeyml(spcs,rxns,solvent=None,solventData=None,path="chem.yml"):
+    D = getmechdict(spcs,rxns,solvent=solvent,solventData=solventData)
+    yaml.dump(D,stream=file(path,'w'))
+
+def getmechdict(spcs,rxns,solvent='solvent',solventData=None):
+    D = dict()
+    D["Units"] = dict()
+    D["Reactions"] = []
+    D["Phases"] = [dict()]
+    D["Phases"][0]["name"] = "phase"
+    D["Phases"][0]["Species"] = [obj2dict(x,spcs) for x in spcs]
+    D["Reactions"] = [obj2dict(x,spcs) for x in rxns]
+    if solventData:
+        D["Solvents"]= [obj2dict(solventData,spcs,label=solvent)]
+    return D
+
+def getradicals(spc):
+    if spc.molecule[0].toSMILES() == "[O][O]":
+        return 0
+    else:
+        return spc.molecule[0].multiplicity-1
+
+def obj2dict(obj,spcs,label="solvent"):
+    D = dict()
+    if isinstance(obj,Species):
+        D["name"] = obj.label
+        D["type"] = "Species"
+        D["smiles"] = obj.molecule[0].toSMILES()
+        D["thermo"] = obj2dict(obj.thermo,spcs)
+        if D["smiles"]  == "[O][O]":
+            D["radicalelectrons"] = 0
+        else:
+            D["radicalelectrons"] = obj.molecule[0].multiplicity-1
+    elif isinstance(obj,NASA):
+        D["polys"] = [obj2dict(k,spcs) for k in obj.polynomials]
+        D["type"] = "NASA"
+    elif isinstance(obj,NASAPolynomial):
+        D["type"] = "NASApolynomial"
+        D["coefs"] = obj.coeffs.tolist()
+        D["Tmax"] = obj.Tmax.value_si
+        D["Tmin"] = obj.Tmin.value_si
+    elif isinstance(obj,Reaction):
+        D["reactants"] = [x.label for x in obj.reactants]
+        D["products"] = [x.label for x in obj.products]
+        D["kinetics"] = obj2dict(obj.kinetics,spcs)
+        D["type"] = "ElementaryReaction"
+        D["radicalchange"] = sum([getradicals(x) for x in obj.products]) -  sum([getradicals(x) for x in obj.reactants])
+    elif isinstance(obj,Arrhenius):
```

Comment:
Is support for the surface kinetics types necessary? I assume not now, since RMG doesn't have surface reactors?

Reply:
I've added them in, RMS can read the catalysis output files properly yet, but it will be able to in the future.